src/server/owned_resp_expr.h

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+#pragma once
+
+#include <variant>
+
+#include "facade/resp_expr.h"
+
+namespace dfly {
+
+class OwnedRespExpr {
+ public:
+  using Buffer = std::vector<uint8_t>;
+
+  using Type = facade::RespExpr::Type;
+
+  using Vec = std::vector<OwnedRespExpr>;
+  Type type;
+
+  std::variant<int64_t, double, Buffer, Vec> u;
```

Comment:
I still don't understand why you can't extedn RespExpr with an OwnedBuffer type and then adapt `TakeRespReply` to conditionally use either of those